src/hw-2/index.ts

```diff
@@ -0,0 +1,75 @@
+import {getAutoSuggestUsers, getUserIndex} from './utils';
+import {User} from './types';
+import {validation} from './validation';
+import * as core from 'express-serve-static-core';
+
+const express = require('express');
+const app = express();
+const router: core.Router = express.Router();
+
+const PORT = 3000;
+const {v1: uuidv4} = require('uuid');
+const userStore: User[] = [];
+
+app.listen(PORT, () => {
+    console.log(`Server is started on port ${PORT}`)
+});
+
+app.use(express.json());
+
+router.route('/user/:id')
+    .get((req: any, res: any, next: any) => {
+        const user: User = userStore.find((item: User) => item.id === req.params.id);
+
+        Boolean(user && !user.isDeleted)
+            ? res.json(user)
+            : res.status(404).json({message: `User with id ${req.params.id} is not found`});
+    })
+    .delete((req: any, res: any) => {
+        const userIndex: number | null = getUserIndex(userStore, req.params.id);
+
+        if (userStore[userIndex] && !userStore[userIndex].isDeleted) {
+            userStore[userIndex] = {...userStore[userIndex], isDeleted: true};
+            res.json({message: `User with id ${req.params.id} is marked as deleted`})
+        } else {
+            res.status(404).json({message: `User with id ${req.params.id} is not found`});
+        }
+    })
+    .patch((req: any, res: any) => {
```

Comment:
I expected to see PUT for U of CRUD, but okay, let it be. Do you know what the difference between http verbs PUT and PATCH and how they're used?

Reply:
I know.